apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -1351,19 +1355,57 @@ def to_anndata(
             shape = measurement.obsm[key].shape
             if len(shape) != 2:
                 raise ValueError(f"expected shape == 2; got {shape}")
-            matrix = measurement.obsm[key].read((slice(None),) * len(shape)).to_numpy()
-            # The spelling ``sp.csr_array`` is more idiomatic but doesn't exist until Python 3.8
-            obsm[key] = sp.csr_matrix(matrix)
+            if isinstance(measurement.obsm[key], DenseNDArray):
+                matrix = (
+                    measurement.obsm[key].read((slice(None),) * len(shape)).to_numpy()
+                )
+                # The spelling ``sp.csr_array`` is more idiomatic but doesn't exist until Python 3.8
+                obsm[key] = sp.csr_matrix(matrix)
```

Comment:
happy to change this @mojaveazure but it predates this PR and it won't be unit-tested given the changes here since we now select sparse -- just a heads-up ...

Reply:
It's OK -- I made the suggested change -- just be aware:

* The change you suggested (and which I accepted) only happens when `obsm`/`varm` are stored densely
* This PR changes `obsm` and `varm` to be stored sparsely
* The `i_want_my_obsms_and_varms_dense` optional config parameter (defaulting to `false`) is out of scope for this issue/PR

Just be aware of those things is all I ask.

(Moreover, I temporarily did write-dense with your mod & it's fine, so, we're fine.)